eng/Version.Details.xml

```diff
@@ -80,9 +80,9 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>f2b7fe854a0b1f78c04dfc065164d6d61040f5b8</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Net.Compilers.Toolset" Version="3.8.0-3.20422.3" CoherentParentDependency="Microsoft.CodeAnalysis.Razor">
+    <Dependency Name="Microsoft.Net.Compilers.Toolset" Version="3.8.0-2.20403.2" CoherentParentDependency="Microsoft.CodeAnalysis.Razor" Pinned="true">
```

Comment:
There's much less reason to downgrade the compiler toolset here. The important thing is to ensure aspnetcore, arcade, and runtime don't keep fighting.
1. arcade is presently at [3.8.0-3.20429.3](https://github.com/dotnet/arcade/blob/master/eng/Version.Details.xml#L42)
2. aspnetcore is at [3.8.0-2.20428.3](https://github.com/dotnet/aspnetcore/blob/master/eng/Version.Details.xml#L332)
3. runtime is at [3.8.0-2.20403.2](https://github.com/dotnet/runtime/blob/master/eng/Versions.props#L54)

Let's use the newest version (3.8.0-3.20429.3) here and in aspnetcore's master branch. And, remove the `CoherentParentDependency` attribute.

Reply:
@mmitche that'll work won't it❔

Reply:
> Let's use the newest version (3.8.0-3.20429.3) here and in aspnetcore's master branch. And, remove the `CoherentParentDependency` attribute.

Do we want to keep it pinned?

Reply:
Yep

Reply:
I'm not positive why we're using the older 'VS 16.8' versions of Roslyn in aspnetcore. The arcade version is newer because it subscribes to the 'VS Master' channel. This is part of the Maestro++ confusion when it creates aspnetcore ➡️ aspnetcore-tooling dependency update PRs.

So, we either
1. update and trigger our roslyn subscription using `darc` in aspnetcore and the 'VS Master' channel then let that newer version flow into this repo as part of next dependency PR here.
    - As I said in #25545, ping me in Teams if you haven't done the `darc` thing before.
2. go with this pinning but use either arcade or aspnetcore's current version of the dependency rather than downgrade it

@mmitche which do you prefer for our 'master' branches❔

Reply:
I think we should probably be subscribing to VS 16.8 in arcade. That would match the 16.8 SDK and be good until we branch arcade for .NET 5. At that point .NET 6's arcade should subscribe to VS master.

Let me run that update for arcade's subscription.

Reply:
In that case, arcade would downgrade slightly while aspnetcore and aspnetcore-tooling would settle on 3.8.0-2.20428.3 without further squabbling. I like it❕

@TanayParikh then we should keep undoing dowgrades of this package manually and wait for the arcade ➡️ aspnetcore ➡️ aspnetcore-tooling flow to right the ⛵ The downgrades are annoying but pinning would be short-lived and something we'd need to remember to undo.

Once @mmitche's Arcade change is promoted, I'll help you or the next on-call person to move that dependency flow as quickly as we can.

Reply:
> we should keep undoing dowgrades of this package manually and wait for the arcade ➡️ aspnetcore ➡️ aspnetcore-tooling flow to right the ⛵ The downgrades are annoying but pinning would be short-lived and something we'd need to remember to undo.

So is there any need for this PR then? Can I close it out or are we still manually upgrading to `3.8.0-2.20428.3`?

Reply:
Interestingly, dotnet/arcade#6105 triggered after @mmitche's Maestro++ / `darc` change. It shows the 'VS 16.8' channel has `v3.8.0-3.20452.1` packages -- newer than what we're using elsewhere.

`v3.8.0-3.20452.1` is also slightly newer than what Roslyn has in the 'VS Master' channel (`3.8.0-3.20451.4`). @jaredpar is that difference intentional❔

Reply:
Yes, when they branched on monday for 16.8 they associated the 16.8 branch with the with the 16.8 channel. VS Master has moved onto 3.8.0-4***

Reply:
> So is there any need for this PR then?

Let's not take this because @mmitche's `darc` update has already started the arcade ➡️ aspnetcore ➡️ aspnetcore-tooling flow of a version we can all agree on😺